src/app/meetups/add-meetups/time-mask.directive.ts

```diff
@@ -0,0 +1,30 @@
+import { Directive, ElementRef, HostListener, Input } from '@angular/core';
+import { NgControl } from '@angular/forms';
+
+@Directive({
+  selector: 'input[planetTimeMask]'
+})
+export class TimeMaskDirective {
+
+  constructor(private _el: ElementRef) { }
+
+  @HostListener('input', [ '$event' ]) onInputChange(event) {
+    const inputValue = this._el.nativeElement.value;
+    console.log(inputValue);
+    const intValue = inputValue.replace(/[^0-9]*/g, '');
+    let hour = intValue.substring(0, 2);
+    let minute = intValue.substring(2, 4);
+    if (hour.length < 2 && +hour > 2) {hour = ''; }
```

Comment:
Rather than all of these conditions, I think there is a regex pattern you could test against for a valid time string and then reject the input if the match fails.